src/DataCollectors/TraceDataCollector/VanguardCollector/Vanguard.cs

```diff
@@ -0,0 +1,380 @@
+        private const string GlobalEventNamePrefix = "Global\\";
+
+        /// <summary>
+        /// Stop() will use this event to check wheather the stop vanguard commnad
+        /// successful stop the collect command codecoverage.exe.
+        /// </summary>
+        private ManualResetEvent vanguardProcessExitEvent;
+
+        /// <summary>
+        /// To monitor vanguard collect command CodeCoverage.exe process.
+        /// </summary>
+        private Process vanguardProcess;
+
+        /// <summary>
+        /// Session name of the vanguard logger.
+        /// </summary>
+        private string sessionName;
+
+        /// <summary>
+        /// Vanguard configuration file name.
+        /// </summary>
+        private string configurationFileName;
+
+        /// <summary>
+        /// Helper object to manage child process lifetimes
+        /// </summary>
+        private IProcessJobObject processJobObject;
+
+        /// <summary>
+        /// Data collection logger
+        /// </summary>
+        private IDataCollectionLogger logger;
+
+        /// <summary>
+        /// Data collection context
+        /// </summary>
+        private DataCollectionContext context;
+
+        private ICollectorUtility collectorUtility;
+
+        private IVangurdCommandBuilder vangurdCommandBuilder;
+
+        public Vanguard()
+            : this(new CollectorUtility(), new VangurdCommandBuilder(), new ProcessJobObject())
+        {
+        }
+
+        internal Vanguard(
+            ICollectorUtility collectorUtility,
+            IVangurdCommandBuilder commandBuilder,
+            IProcessJobObject processJobObject)
+        {
+            this.collectorUtility = collectorUtility;
+            this.vangurdCommandBuilder = commandBuilder;
+            this.processJobObject = processJobObject;
+        }
+
+        /// <summary>
+        /// Gets output file name of vanguard
+        /// </summary>
+        public string OutputName { get; private set; }
+
+        /// <summary>
+        /// Gets a value indicating whether whether vanguard is running
+        /// </summary>
+        private bool IsRunning
+        {
+            get { return this.vanguardProcess != null && !this.vanguardProcess.HasExited; }
+        }
+
+        /// <inheritdoc />
+        public void Initialize(
+            string sessionName,
+            string configurationFileName,
+            XmlElement configuration,
+            IDataCollectionLogger logger)
+        {
+            ValidateArg.NotNull(configuration, nameof(configuration));
+
+            EqtTrace.Info("Vanguard.Initialize: Session name: {0}, config filename: {1} config: {2}", sessionName, configurationFileName, configuration.InnerXml);
+
+            this.sessionName = sessionName;
+            this.configurationFileName = configurationFileName;
+            this.logger = logger;
+            using (var writer = new StreamWriter(new FileStream(this.configurationFileName, FileMode.Create)))
```

Comment:
>FileStream [](start = 53, length = 10)

Please check if FileStream is disposed or not

Reply:
Moved writing to config to configFile logic to Impl. Used `File.WriteAllText` API.